TGM/src/main/java/com/minehut/tgm/command/CycleCommands.java

```diff
@@ -153,8 +154,23 @@ public static void setNext(CommandContext cmd, CommandSender sender) {
     @Command(aliases = {"join"}, desc = "Join a team.")
     public static void join(CommandContext cmd, CommandSender sender) {
         if (cmd.argsLength() == 0) {
-            MatchTeam matchTeam = TGM.get().getModule(TeamManagerModule.class).getSmallestTeam();
-            attemptJoinTeam((Player) sender, matchTeam, true);
+            if (TGM.get().getModule(TeamManagerModule.class).getTeam((Player) sender).isSpectator()) {
+                if (TGM.get().getMatchManager().getMatch().getMapContainer().getMapInfo().getGametype().equals(GameType.Infected)) {
+                    if (TGM.get().getMatchManager().getMatch().getMatchStatus().equals(MatchStatus.MID)) {
+                        MatchTeam team = TGM.get().getModule(TeamManagerModule.class).getTeamById("infected");
+                        attemptJoinTeam((Player) sender, team, true);
+                        return;
+                    }
+
+                    MatchTeam team = TGM.get().getModule(TeamManagerModule.class).getTeamById("humans");
+                    attemptJoinTeam((Player) sender, team, true);
+                    return;
+                }
+                MatchTeam matchTeam = TGM.get().getModule(TeamManagerModule.class).getSmallestTeam();
+                attemptJoinTeam((Player) sender, matchTeam, true);
+            } else {
+                sender.sendMessage(ChatColor.RED + "You have already chosen a team.");
+            }
```

Comment:
Use some local variables instead of constantly searching for the modules on each line.